### PR TITLE
Implement build-iso

### PR DESCRIPTION
**Pull request by fepitre**
With several fixes, it allows to create a workflow for building testing ISO. It intends to be used for weekly builds in order to
reproduce stable/official releases build environment. Moreover, if openqa is configured, it would trigger tests suite.

### CODE REVIEW

File: github-action.py [hunk 1]
```diff
@@ -93,7 +100,7 @@ class CommitMismatchError(AutoActionError):
     pass
 
 
-class BaseAutoAction(ABC):
+class BaseAutoAction:
```
Comment (marmarek) [inline github-action.py]: Why not ABC? If you don't need `upload` in ISO action, you can add dummy method (raising NotImplementedError) instead.

File: github-action.py [hunk 2]
```diff
@@ -740,6 +746,211 @@ def upload(self):
...
+            commit_sha=commit_sha,
+            repository_publish=repository_publish,
+            local_log_file=local_log_file,
+        )
+
+        self.timeout = self.config.get("timeout", 21600)
+
+        config_iso = self.config.get("iso", {})
+        config_iso["version"] = commit_sha
+        self.config.set("iso", config_iso)
```
Comment (marmarek) [inline github-action.py:779]: There should be also some way to set `is-final` (for example when building RC vs final release). I'm okay-ish with doing that implicitly based on the version (whether it contains rc/alpha/beta) - for example configurable list of strings to look in version to force is-final=false.

File: github-action.py [hunk 3]
```diff
@@ -740,6 +746,211 @@ def upload(self):
...
+                )
+
+                stage = "upload"
+                self.make_with_log(
+                    self.run_stages,
+                    stages=["upload"],
+                )
+
+                additional_info = None
+                if OpenQA_Client and OpenQAClientError:
```
Comment (marmarek) [inline github-action.py]: This should be configurable, not only by the module presence.

**Review by marmarek — CHANGES_REQUESTED** [review]

File: github-action.py (hunk 1 above)
Comment (fepitre) [reply github-action.py]: Yes I did at first, just wondering if ABC were really necessary but yes let's do this dummy method.

File: github-action.py [hunk 4]
```diff
@@ -740,6 +750,226 @@ def upload(self):
...
+    def __init__(
+        self,
+        builder_dir,
+        config,
+        iso_timestamp,
+        state_dir,
+        commit_sha,
+        repository_publish,
+        local_log_file,
+        is_final=False,
```
Comment (marmarek) [inline github-action.py:763]: How `is_final` can be set? Currently only via a manual call, right?

File: github-action.py [hunk 5]
```diff
@@ -740,6 +750,226 @@ def upload(self):
...
+            ]
+
+        try:
+            subprocess.run(notify_issues_cmd, env=self.env)
+        except subprocess.CalledProcessError as e:
+            msg = f"{package_name}: Failed to notify GitHub: {str(e)}"
+            log.error(msg)
+
+    def upload(self):
+        raise NotImplemented
```
Comment (marmarek) [inline github-action.py]:
```suggestion
        raise NotImplementedError
```
`NotImplemented` exists too, but is not an exception

**Review by marmarek — CHANGES_REQUESTED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**